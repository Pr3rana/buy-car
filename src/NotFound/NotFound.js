import React from 'react';
import {Link} from 'react-router-dom';
import './NotFound.css';
import logo from '../logo.png';

const NotFound = ()=>{
    return(
        <div className="error-main">
            <img alt="logo" className="logo" src={logo}/>
            <p className="error-type">404 - Not Found</p>
            <p>
                Sorry, the page you are looking for doesn't exist. <br/>
                You can always go back to <Link to = '/'>homepage</Link>.
            </p>
        </div>
    );
}
export default NotFound;
