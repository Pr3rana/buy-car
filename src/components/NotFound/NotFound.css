
.error-main{
    position: relative;
    top: 15vh;
    text-align: center;
    font-weight: 400;
    font-size: 18px;
}

.error-main a{
    color: #ea7f28;
    text-decoration: none;
    display: inline-block;
}
.error-main a:hover, .error-main a:active{
    color: #d37324;
    border-bottom: 1px solid #d37324;
}

.error-type{
    font-weight: 800;
    font-size: 32px;
    color: #4a4a4a;
}