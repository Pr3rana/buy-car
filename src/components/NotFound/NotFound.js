import React from 'react';
import './NotFound.css';
import logo from '../../logo.png';

const NotFound = ({errorType="Error", errorDetails=null, brandLogo = logo})=>{
    return(
        <div data-testid = "not-found" className="error-main">
            <img alt="logo" className="logo" src={brandLogo}/>
            <p className="error-type">{errorType}</p>
            {errorDetails}
        </div>
    );
}
export default NotFound;
