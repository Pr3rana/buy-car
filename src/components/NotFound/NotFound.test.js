import { render, screen, cleanup } from '@testing-library/react';
import NotFound from './NotFound';

afterEach(cleanup);
const errorDetails = <p>Sorry, the page you are looking for doesn't exist. <br/>You can always go back to <a href='/'>homepage</a>.</p>
describe("NotFound render", ()=>{
    test('Initial Render', () => {
        const { container } = render(<NotFound />)
        expect(container.firstChild).toMatchSnapshot(`
        <div className="error-main">
            <img alt="logo" className="logo" src="../../logo.png"/>
            <p className="error-type">Error</p>
        </div>
        `)
        expect(screen.queryByTestId('NotFound-list-container')).not.toBeInTheDocument()
    });
});

describe("NotFound props",()=>{
    test("Re-render the NotFound with different props",()=>{
        const { container } = render(<NotFound errorType="404 - Not Found" errorDetails={errorDetails}/>)
        expect(screen.getByTestId('not-found')).toHaveTextContent('404 - Not Found');
        expect(container.firstChild).toMatchSnapshot(`
        <div className="error-main">
            <img alt="logo" className="logo" src={brandLogo}/>
            <p className="error-type">404 - Not Found</p>
            <p>Sorry, the page you are looking for doesn't exist. <br/>You can always go back to <a href='/'>homepage</a>.</p>
        </div>
        `)
    })
})