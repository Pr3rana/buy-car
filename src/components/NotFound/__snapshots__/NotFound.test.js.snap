// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound props Re-render the NotFound with different props: 
        <div className="error-main">
            <img alt="logo" className="logo" src={brandLogo}/>
            <p className="error-type">404 - Not Found</p>
            <p>Sorry, the page you are looking for doesn't exist. <br/>You can always go back to <a href='/'>homepage</a>.</p>
        </div>
         1`] = `
<div
  class="error-main"
  data-testid="not-found"
>
  <img
    alt="logo"
    class="logo"
    src="logo.png"
  />
  <p
    class="error-type"
  >
    404 - Not Found
  </p>
  <p>
    Sorry, the page you are looking for doesn't exist. 
    <br />
    You can always go back to 
    <a
      href="/"
    >
      homepage
    </a>
    .
  </p>
</div>
`;

exports[`NotFound render Initial Render: 
        <div className="error-main">
            <img alt="logo" className="logo" src="../../logo.png"/>
            <p className="error-type">Error</p>
        </div>
         1`] = `
<div
  class="error-main"
  data-testid="not-found"
>
  <img
    alt="logo"
    class="logo"
    src="logo.png"
  />
  <p
    class="error-type"
  >
    Error
  </p>
</div>
`;
